modules/api/pkg/ee/clusterbackup/backupstoragelocation/handler.go

```diff
@@ -0,0 +1,458 @@
+
+   END OF TERMS AND CONDITIONS
+*/
+
+package backupstoragelocation
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/gorilla/mux"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+
+	apiv2 "k8c.io/dashboard/v2/pkg/api/v2"
+	clusterbackup "k8c.io/dashboard/v2/pkg/ee/clusterbackup/backup"
+	handlercommon "k8c.io/dashboard/v2/pkg/handler/common"
+	"k8c.io/dashboard/v2/pkg/handler/v1/common"
+	"k8c.io/dashboard/v2/pkg/handler/v2/cluster"
+	"k8c.io/dashboard/v2/pkg/provider"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/log"
+	utilerrors "k8c.io/kubermatic/v2/pkg/util/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	displayNameLabelKey                  = "bsl-display-name"
+	credentialsSecretKeyName             = "cloud-credentials"
+	clusterIdLabelKey                    = "cluster-id"
+	importBSLNamePrefix                  = "import-bsl"
+	bslRegion                            = "region"
+	bslS3Url                             = "s3Url"
+	bslKind                              = "BackupStorageLocation"
+	defaultCloudCredentialsSecretKeyName = "cloud"
+)
+
+var credentialsTemplate string = `[default]
+aws_access_key_id = {{ .awsAccessKeyId }}
+aws_secret_access_key = {{ .awsSecretAccessKey }}
+`
+
+// createBSLReq defines HTTP request to create bsl which would be used to restore backups from different cluster
+// swagger:parameters createBackupStorageLocation
+type createBSLReq struct {
+	cluster.GetClusterReq
+
+	// in: body
+	// required: true
+	Body BslBody
+}
+
+type BslBody struct {
+	BSLSpec velerov1.BackupStorageLocationSpec `json:"bslSpec,omitempty"`
+}
+
+func DecodeCreateBSLReq(c context.Context, r *http.Request) (interface{}, error) {
+	var req createBSLReq
+	cr, err := cluster.DecodeGetClusterReq(c, r)
+
+	if err != nil {
+		return nil, err
+	}
+
+	req.GetClusterReq = cr.(cluster.GetClusterReq)
+
+	if err = json.NewDecoder(r.Body).Decode(&req.Body); err != nil {
+		return nil, err
+	}
+	return req, nil
+}
+
+func (r *createBSLReq) validateCreateBSLReq() error {
+	bslSpec := r.Body.BSLSpec
+	if bslSpec.Provider != "aws" {
+		return fmt.Errorf("velero plugin not supported other than aws")
+	}
+	return nil
+}
+
+func CreateBSLEndpoint(ctx context.Context, request interface{}, userInfoGetter provider.UserInfoGetter, backupProvider provider.BackupStorageProvider, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, settingsProvider provider.SettingsProvider) (*apiv2.BackupStorageLocation, error) {
+	if err := clusterbackup.IsClusterBackupEnabled(ctx, settingsProvider); err != nil {
+		return nil, err
+	}
+
```

Comment:
This feature and all the endpoints associated with it need extra guardrails to make sure that it's only accessible for cluster admins or project admins.